### DISCUSSION TITLE
Import existing Maven projects: remove root directory suggestion

### DISCUSSION THREAD

**Discussion by UlrichScholz** [eclipse-m2e/m2e-core#1602] [Q&A] [answered]
When importing existing Java Maven projects with File > Import > Existing Maven Projects, Eclipse provides suggestions under Root Directory.

How can I remove such a suggestion (and keep the others)?

Eclipse IDE for Enterprise Java and Web Developers (includes Incubating components)
Version: 2023-06 (4.28.0)
Build id: 20230608-1333
OS: Windows 11, v.10.0, x86_64 / win32
Java vendor: Oracle Corporation
Java runtime version: 17.0.7+8-LTS-224
Java version: 17.0.7

**Comment by mickaelistria** [comment]
Have you tried File > Open Projects from filesystem?

**Reply by UlrichScholz** [reply]
Mmmh, no. But that's probably not what I want. Import > Existing Maven Projects and selecting the correct Root Directory does the right thing.

The point of my question is: Eclipse is suggesting wrong Root Directories.  I want to remove these wrong suggestions from the list.

**Comment by mickaelistria** [comment]
> But that's probably not what I want

Have you tried? What does it do, or not, that you don't want?

**Reply by UlrichScholz** [reply]
I just tried. Import > Existing Maven Projects and selecting the correct Root Directory gives a list of existing Maven projects. That's probably the reason this option is called "Existing Maven Projects". 
Import Projects from File System with Input Source set to the same root directory gives a list of plugins. This list differs from the list given by Import > Existing Maven Projects by count and name of elements. I would have to try to find the correct set.  And then, will it work? So why using that when Import > Existing Maven Projects does the right thing?

**Comment by laeubi** [comment]
@UlrichScholz `Import > Existing Maven Projects` is usually the right choice, but maybe you can give a public project to try and share some screenshot its quite easy with github, just copy the image (or a portion of it with your favorite image tools) and STRG+V it into the comment.

I don't really understand the problem and why one should "remove the root suggestion"....

**Reply by UlrichScholz** [reply]
I don't think I can provide an example with a public project. But I can share a screen shot:

- `C:\SmartShift\D1_smarttools` is the correct entry
- `C:\SmartShift\D1_smarttools\plugins` is the unwanted one. Some time ago, it was required. I probably entered it via Browse the first time it was required. Then, it ended up as suggestion. Now, it is no longer required.

![image](https://github.com/eclipse-m2e/m2e-core/assets/13901484/c42b23cc-1821-48dc-9272-1a5ae3aae585)

BTW, both suggestions are still valid file paths.

**Reply by laeubi** [reply]
Cna you explain why second is "unwanted" / "invalid" and how m2e should detect such intention? I think this is just a history of recently entered pathes...

**Reply by UlrichScholz** [reply]
It is unwanted because the user (read: I) do not want it anymore to be shown. It has no meaning in the project's Maven setup anymore.
I do not expect/request m2e to detect that, nor to do something automatically. All I want is some advice how I can remove that option from the list by doing some manual steps on my machine. It would be fine for me to remove all suggestions at once. Then, I would use Browse to select `C:\SmartShift\D1_smarttools` once again, it would be recorded as suggestion, and I would live happily ever after.

**Reply by laeubi** [reply, marked as answer]
@UlrichScholz this is not provided by m2e this is part of the smart import of eclipse platform IDE:
https://github.com/eclipse-platform/eclipse.platform.ui/ and stored in the dialog settings of JFace

As far as I can see this is save in your workspace in the plugin area so probabbaly you can search for the string on your workspace stored path and find the file and remove it then...

**Reply by UlrichScholz** [reply]
Yes, it is a value in `.metadata\.plugins\org.eclipse.m2e.core.ui\dialog_settings.xml`. Really hard to find. Thanks
